ompi/datatype/ompi_datatype_internal.h

```diff
@@ -99,8 +99,14 @@
  */
 #define OMPI_DATATYPE_MPI_COUNT                   0x2E
 
+/*
+ * Datatypes proposed in the MPI Forum
```

Comment:
Minor quibble: please put a date / context here for future code readers to know the scope.  E.g.:

> Datatypes proposed to the MPI Forum in December 2018 for proposal in the MPI v4.0 specification.

The reason for this is because if we forget to update this when the FP16 datatypes are accepted / rejected by the MPI Forum, some future dev who is reading this comment has enough context to dig back into the MPI Forum v4.0 spec and/or Forum history to know whether this is stale code or whether it needs to be converted from MPIX -> MPI.